packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1006,11 +1006,39 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
   }
 
   void _resolveImage() {
-    final ImageStream newStream =
-      widget.image.resolve(createLocalImageConfiguration(
+    final DeferringImageProvider<dynamic> provider = DeferringImageProvider<dynamic>(
+      imageProvider: widget.image,
+      getNextAction: () {
+        // Did this element get disposed of (scrolled too far out of view before
+        // scrolling slowed down enough to catch up)?
+        if (!mounted) {
+          return DeferringImageProviderAction.cancel;
+        }
+        // Is it scrolling so fast that the image is unlikely stay around long
+        // enough to be worth decoding? If so, wait until the next task loop
+        // to see if we've slowed down or disappeared.
+        // We're using the large of height or width of the window in physical
+        // pixels. This operation compares physical pixels (the window
+        // dimension) with logical pixels (the scrolling velocity). Using
+        // logical window pixels results in being too aggressive about loading
+        // images, and physical pixels will give us a better approximation of
+        // how much RAM is available.
+        // It's safe to use these calls as point-in-time - do not subscribe to
+        // updates of MediaQuery or Scrollable.
+        final double maxPhysicalPixels = WidgetsBinding.instance.window.physicalSize.longestSide;
+        if ((Scrollable.scrollingVelocityOfContext(context) ?? 0).abs() > maxPhysicalPixels) {
```

Comment:
It is a little surprising to tie our Image implementation to our implementation of Scrollable. If somebody were to write a new Scrolling lib (because they don't like our sliver-based on or whatever) they would not be able to make uses of this feature for images shown in their scrolling container... This my be more hypothetical, though.

Reply:
Yes, I agree this is a problem.  I'm not sure what we can do about it at the moment. I think we should punt on it for now and deal with it if it comes up - we could rework this so that you could find other ways to notify widgets about how fast you are scrolling.